warehouse/admin/views/checks.py

```diff
@@ -43,19 +43,61 @@ def get_checks(request):
     uses_session=True,
 )
 def get_check(request):
-    query = (
+    check = get_check_by_name(request.db, request.matchdict["check_name"])
+
+    all_checks = (
         request.db.query(MalwareCheck)
         .filter(MalwareCheck.name == request.matchdict["check_name"])
         .order_by(MalwareCheck.version.desc())
+        .all()
     )
 
+    return {"check": check, "checks": all_checks, "states": MalwareCheckState}
+
+
+@view_config(
+    route_name="admin.checks.run_backfill",
+    permission="admin",
+    request_method="POST",
+    uses_session=True,
+    require_methods=False,
+    require_csrf=True,
+)
+def run_backfill(request):
+    check = get_check_by_name(request.db, request.matchdict["check_name"])
+
     try:
-        # Throw an exception if and only if no results are returned.
-        newest = query.limit(1).one()
-    except NoResultFound:
+        sample = int(request.POST["sample-rate"])
+    except (KeyError, ValueError):
         raise HTTPNotFound
 
-    return {"check": newest, "checks": query.all(), "states": MalwareCheckState}
+    if check.state not in (MalwareCheckState.enabled, MalwareCheckState.evaluation):
+        request.session.flash(
+            f"Check must be in 'enabled' or 'evaluation' state to run a backfill.",
+            queue="error",
+        )
+        return HTTPSeeOther(
+            request.route_path("admin.checks.detail", check_name=check.name)
+        )
+
+    if sample <= 0 or sample > 100:
```

Comment:
It might make sense to have a dedicated form for validation here, instead of performing the validation in the request handler.  It looks like the admin views don't follow the same pattern of dedicated forms, though, so feel free to ignore.

Reply:
I do think it makes sense to implement the form validation found in the user-facing site, but maybe not in this PR. This will be needed when I add advanced filtering capabilities in the verdicts view, so I'll add it here then as well.

Reply:
It's a mix in admin for sure, but see `admin/views/users.py` for examples of a form with validators (EmailForm and UserForm)